app/src/main/java/de/danoeh/antennapod/preferences/PreferenceController.java

```diff
@@ -433,9 +436,17 @@ public void onResume() {
         setParallelDownloadsText(UserPreferences.getParallelDownloads());
         setEpisodeCacheSizeText(UserPreferences.getEpisodeCacheSize());
         setDataFolderText();
+        GpodnetPreferences.setSyncAttemptListener(() -> ui.getActivity().runOnUiThread(
```

Comment:
This really looks like you re-invented ``SharedPreferences.OnSharedPreferenceChangeListener``.